sphinx/environment/__init__.py

```diff
@@ -91,6 +92,18 @@
 }  # type: Dict[unicode, Union[bool, Callable]]
 
 
+@contextmanager
+def sphinx_smartquotes_action(env):
+    try:
+        original = SphinxSmartQuotes.smartquotes_action
+        setattr(SphinxSmartQuotes, 'smartquotes_action',
```

Comment:
Is there any reason to using `setattr`? An assignment is much simpler to me.

Reply:
I used an assignment but it did not have expected effect in my testing. Was it because SphinxSmartQuotes is imported separately in io.py?

Reply:
hmm... works perfectly well with an assignment. Don't know what went wrong (but I am doing various things at the same time, bad habit...). Ok, will push fix.